### ISSUE TITLE
ambiguous behaviour of docstring-min-length

### ISSUE THREAD

**Issue by mcallaghan-bsm** [PyCQA/pylint#2946]
### Steps to reproduce
1. Leverage `docstring-min-length` as per https://pylint.readthedocs.io/en/latest/technical_reference/features.html#basic-checker-options in a `.pylintrc` file.
2. Set minimum line length to `10`

### Current behavior
 * the presence of `docstring-min-length` seems to nullify the requirement to have ANY docstring at all. (regardless of `no-docstring-rgx`)
 * documentation states:
> Minimum line length for functions/classes that require docstrings, shorter ones are exempt. Default: -1

So consider this fabricated module with some functions in it
``` python
"""a dummy module for testing pylint docstring tunables"""

def print_pets(cat, dog, fish):
    print(cat, dog, fish)


def print_animals(horse, goat, coyote):
    print(horse, goat, coyote)

    if horse:
        print('we have a horse')
    elif goat:
        print('we have a goat')
    elif coyote:
        print('we have a coyote')
    else:
        raise ValueError('an unlikely situation to not have any of these')
```

So we have TWO methods defined, neither of them have docstrings. One is moderately "complex", the other simpler.

And the following `.pylintrc` (snipped relevant items from the default `--generaterc`)
``` ini

[BASIC]

# Regular expression which should only match function or class names that do
# not require a docstring.
#
no-docstring-rgx=^_

# Minimum line length for functions/classes that require docstrings, shorter
# ones are exempt.
#
#docstring-min-length=10
```

So to start, it behaves correctly, telling us that the methods are missing docstring.
```
# pipenv run pylint --rcfile=.pylintrc_docstring test_docstring_pylint.py   | grep docstring
************* Module test_docstring_pylint
test_docstring_pylint.py:3:0: C0111: Missing function docstring (missing-docstring)
test_docstring_pylint.py:7:0: C0111: Missing function docstring (missing-docstring)
```

But as soon as we enable the `min-length` requirement (uncomment it)
```
# pipenv run pylint --rcfile=.pylintrc_docstring test_docstring_pylint.py   | grep docstring
************* Module test_docstring_pylint
test_docstring_pylint.py:7:0: C0111: Missing function docstring (missing-docstring)
```

Why is `print_pets` no longer included?  It is missing docstring ENTIRELY.  It should still have the warning.  The docs ambiguously state "shorter ones are exempt", this is reasonable to exclude the min-length requirement for simple functions, but not MISSING.  (feels like a defect to me).

Now if we add a simple docstring to `print_animals`,
``` python
def print_animals(horse, goat, coyote):
    """Prints some animals"""
```

And now neither `print_pets` nor `print_animals` complain ... the second part of this issue is how does pylint conclude "short ones" for exclusion?
```
pipenv run pylint --rcfile=.pylintrc_docstring test_docstring_pylint.py   | grep docstring
(EMPTY)
```

### Expected behavior

1. Clarification (explanation) here, and into documentation to get rid of the ambiguity for how this configuration is meant to behave.
2. `docstring-min-length` should not render methods/functions to be allowed to not have ANY docstring.
3. at what point / scenario does `min-length` come into play? (can't get it to work at all)

### pylint --version output
```
# pipenv run pylint --version
pylint 2.3.1
astroid 2.2.5
Python 3.7.3 (default, May  8 2019, 05:31:59) 
[GCC 6.3.0 20170516]
```

**Comment by PCManticore** [comment]
Thanks for the clear explanation. I agree this behaviour doesn't make that much sense and we should fix it.

**Comment by earonesty** [comment]
Weird that this is a bug.  It seems to be working-as-designed.   If you set the minimum length, shorter functions are exempt from the requirement to have docstrings.

**Comment by DanielNoord** [comment]
Yeah this is working as intended. `docstring-min-length` refers to the length of the function or class itself:
Minimum **line length for functions/classes that require docstrings**, shorter ones are exempt.
